File: agent-mesh-py/python/agent_mesh/amesh/whoami.py
"""``amesh whoami`` — print the local user identity.

Mirror of ``agent-mesh-cli/src/whoami.rs``. Loads ``<home>/user.key``,
prints its fingerprint, then reports the GitHub binding state from
``<home>/user.github.sig`` (if present).
"""

from __future__ import annotations

import json
import sys
from pathlib import Path

from agent_mesh.core import UserKey


def run(home: Path) -> int:
    """Print user fingerprint + github binding hint.

    Returns 0 on success, 1 if no user key exists yet.
    """
    key_path = home / "user.key"
    if not key_path.exists():
        print(
            f"error: load {key_path} — run `amesh keygen` first?",
            file=sys.stderr,
        )
        return 1

    key = UserKey.load(str(key_path))
    fp = key.fingerprint()
    print(f"user fingerprint: {fp.hex()}")
    print(f"short:            {fp.short()}")

    binding_path = home / "user.github.sig"
    if binding_path.exists():
        # We only need the optional `github_username` field for the
        # hint — read the raw JSON rather than rebuilding the binding
        # object, so whoami doesn't fail loud on a partially-corrupt
        # binding file.
        try:
            data = json.loads(binding_path.read_text())
            username = data.get("github_username")
        except (json.JSONDecodeError, OSError):
            username = None
        if username:
            print(f"github binding:   {username} (hint)")
        else:
            print("github binding:   (no username hint)")
    else:
        print("github binding:   none (run `amesh bind github` to add one)")
    return 0

File: agent-mesh-cli/src/whoami.rs
//! `amesh whoami` — print the local user identity.

use agent_mesh_protocol::UserKey;
use anyhow::{Context, Result};
use std::path::PathBuf;

/// Load the user key from `<home>/user.key` and print its
/// fingerprint, plus any GitHub binding hint we find alongside it.
pub fn run(home: PathBuf) -> Result<()> {
    let key_path = home.join("user.key");
    let key = UserKey::load(&key_path)
        .with_context(|| format!("load {} — run `amesh keygen` first?", key_path.display()))?;
    let fp = key.fingerprint();
    println!("user fingerprint: {}", fp.hex());
    println!("short:            {}", fp.short());

    let binding_path = home.join("user.github.sig");
    if binding_path.exists() {
        let json = std::fs::read_to_string(&binding_path)?;
        let binding: agent_mesh_protocol::GitHubBinding = serde_json::from_str(&json)?;
        match binding.github_username.as_deref() {
            Some(u) => println!("github binding:   {u} (hint)"),
            None => println!("github binding:   (no username hint)"),
        }
    } else {
        println!("github binding:   none (run `amesh bind github` to add one)");
    }

    Ok(())
}
